sbt-plugin/src/main/scala/scala/scalajs/sbtplugin/ScalaJSJUnitPlugin.scala

```diff
@@ -0,0 +1,43 @@
+/*                     __                                               *\
+**     ________ ___   / /  ___      __ ____  Scala.js sbt plugin        **
+**    / __/ __// _ | / /  / _ | __ / // __/  (c) 2013, LAMP/EPFL        **
+**  __\ \/ /__/ __ |/ /__/ __ |/_// /_\ \    http://scala-js.org/       **
+** /____/\___/_/ |_/____/_/ | |__/ /____/                               **
+**                          |/____/                                     **
+\*                                                                      */
+
+package org.scalajs.sbtplugin
+
+import sbt._
+import sbt.Keys._
+
+object ScalaJSJUnitPlugin extends AutoPlugin {
+  override def requires: Plugins = ScalaJSPlugin
+
+  override def projectSettings: Seq[Setting[_]] = {
+    ScalaJSPluginInternal.scalaJSAbstractSettings ++
+      ScalaJSPluginInternal.scalaJSEcosystemSettings ++
+      Seq(
+        ivyConfigurations += config("scala-js-test-plugin").hide,
+        libraryDependencies ++= Seq(
+          "org.scala-js" % "scalajs-junit-test-plugin" %
+            ScalaJSPlugin.autoImport.scalaJSVersion %
+            "scala-js-test-plugin" cross CrossVersion.full,
+          "org.scala-js" %% "scalajs-junit-test-runtime" %
+            ScalaJSPlugin.autoImport.scalaJSVersion  % "test"
+        ),
+        scalacOptions in Test ++= {
+          val report = update.value
+          val jars = report.select(configurationFilter("scala-js-test-plugin"))
+          for {
+            jar <- jars
+            jarPath = jar.getPath
+            // This is a hack to filter out the dependencies of the plugins
+            if jarPath.contains("plugin")
+          } yield {
+            s"-Xplugin:$jarPath"
+          }
+        }
```

Comment:
`addCompilerPlugin` enables the plugin in both `Compile` and `Test`. This only enables it in the `Test` configuration. We don't want to pollute the main artifact with code that has been tampered with by an (additional) compiler plugin.